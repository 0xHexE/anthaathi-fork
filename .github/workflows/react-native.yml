name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
