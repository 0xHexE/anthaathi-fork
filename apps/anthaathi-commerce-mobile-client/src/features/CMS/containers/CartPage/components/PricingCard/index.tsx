import {View, Text, TextStyle, StyleProp} from 'react-native';
import React from 'react';
import {Card} from 'react-native-paper';
import {useIntl} from 'react-intl';
import {CartPageComponentType} from '../../../../types/common';

export interface ViewTextProps {
  title: string;
  subtitle: string;
  titleStyle?: StyleProp<TextStyle>;
  subtitleStyle?: StyleProp<TextStyle>;
}

export interface PriceCurrencyProps {
  currency: string;
  price: number;
}
export interface PricingCardProps {
  subtotal: PriceCurrencyProps;
  discount: PriceCurrencyProps;
  promoDiscount: PriceCurrencyProps;
  shippingCharges: PriceCurrencyProps;
  total: PriceCurrencyProps;
}

const PricingCard = (props: PricingCardProps) => {
  const intl = useIntl();

  return (
    <Card
      style={{
        marginHorizontal: 10,
        borderColor: '#E3E2E7',
        borderWidth: 1,
        borderRadius: 4,
        marginVertical: 5,
      }}
      testID="pricingCard">
      <Card.Content>
        <ViewText
          title={intl.formatMessage({defaultMessage: 'Subtotal'})}
          subtitle={intl.formatNumber(props.subtotal.price, {
            style: 'currency',
            currency: props.subtotal.currency,
          })}
        />
        <ViewText
          title={intl.formatMessage({defaultMessage: 'Discount'})}
          subtitle={intl.formatNumber(props.discount.price, {
            style: 'currency',
            currency: props.discount.currency,
          })}
        />
        <ViewText
          title={intl.formatMessage({defaultMessage: 'Promo Discount'})}
          subtitle={intl.formatNumber(props.promoDiscount.price, {
            style: 'currency',
            currency: props.promoDiscount.currency,
          })}
        />
        <ViewText
          title={intl.formatMessage({defaultMessage: 'Shipping Charges.'})}
          subtitle={intl.formatNumber(props.shippingCharges.price, {
            style: 'currency',
            currency: props.shippingCharges.currency,
          })}
          subtitleStyle={{fontWeight: '500', color: '#E94040'}}
        />
        <ViewText
          title={intl.formatMessage({defaultMessage: 'Total'})}
          subtitle={intl.formatNumber(props.total.price, {
            style: 'currency',
            currency: props.total.currency,
          })}
          titleStyle={{fontSize: 16, fontWeight: '700', color: '#364A15'}}
          subtitleStyle={{fontSize: 16, fontWeight: '700', color: '#008D3E'}}
        />
      </Card.Content>
    </Card>
  );
};

const ViewText = ({
  title,
  subtitle,
  titleStyle,
  subtitleStyle,
}: ViewTextProps) => {
  return (
    <View
      style={{
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginHorizontal: 3,
        marginVertical: 5,
      }}>
      <Text
        testID="title"
        style={[{color: '#808080', fontSize: 14}, titleStyle]}>
        {title}
      </Text>
      <Text
        testID="subtitle"
        style={[
          {color: '#364A15', fontSize: 14, fontWeight: '600'},
          subtitleStyle,
        ]}>
        {subtitle}
      </Text>
    </View>
  );
};

export default PricingCard;

export const PricingCardCMSInput = {
  _component: CartPageComponentType.PricingCard,
  component: PricingCard,
};
