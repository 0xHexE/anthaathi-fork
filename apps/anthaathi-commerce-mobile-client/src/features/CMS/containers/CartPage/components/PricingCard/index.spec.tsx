import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PricingCard from './index';

describe('PricingCard', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PricingCard
            subtotal={{currency: 'USD', price: 10.1}}
            discount={{currency: 'USD', price: 10.1}}
            promoDiscount={{currency: 'USD', price: 10.1}}
            shippingCharges={{currency: 'USD', price: 0}}
            total={{currency: 'USD', price: 10.1}}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('pricingCard')).toBeTruthy();
  });

  it('should have PricingCard title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PricingCard
            subtotal={{currency: 'USD', price: 10.1}}
            discount={{currency: 'USD', price: 10.1}}
            promoDiscount={{currency: 'USD', price: 10.1}}
            shippingCharges={{currency: 'USD', price: 0}}
            total={{currency: 'USD', price: 10.1}}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryAllByTestId('title').length).toBe(5);
    expect(temp.queryAllByTestId('subtitle').length).toBe(5);
  });
});
