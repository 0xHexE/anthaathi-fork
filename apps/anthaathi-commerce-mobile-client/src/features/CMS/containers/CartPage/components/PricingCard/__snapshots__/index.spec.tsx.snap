// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard should render the item 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "borderColor": "#E3E2E7",
          "borderRadius": 4,
          "borderWidth": 1,
          "elevation": 1,
          "marginHorizontal": 10,
          "marginVertical": 5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="pricingCard"
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 3,
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
              testID="title"
            >
              Subtotal
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  undefined,
                ]
              }
              testID="subtitle"
            >
              $10.10
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 3,
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
              testID="title"
            >
              Discount
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  undefined,
                ]
              }
              testID="subtitle"
            >
              $10.10
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 3,
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
              testID="title"
            >
              Promo Discount
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  undefined,
                ]
              }
              testID="subtitle"
            >
              $10.10
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 3,
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
              testID="title"
            >
              Shipping Charges.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#E94040",
                    "fontWeight": "500",
                  },
                ]
              }
              testID="subtitle"
            >
              $0.00
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 3,
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#364A15",
                    "fontSize": 16,
                    "fontWeight": "700",
                  },
                ]
              }
              testID="title"
            >
              Total
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#008D3E",
                    "fontSize": 16,
                    "fontWeight": "700",
                  },
                ]
              }
              testID="subtitle"
            >
              $10.10
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
